File: backend/app/schemas/ai_monitor.py
from pydantic import BaseModel, UUID4
from datetime import datetime
from typing import List, Optional

class AIMetricBase(BaseModel):
    module: Optional[str] = None
    model: str
    provider: str
    latency: float
    prompt_tokens: int
    completion_tokens: int
    total_tokens: int
    status: str
    error_message: Optional[str] = None

class AIMetricOut(AIMetricBase):
    id: UUID4
    timestamp: datetime
    
    class Config:
        from_attributes = True

class PaginatedAIMetrics(BaseModel):
    total: int
    items: List[AIMetricOut]
    page: int
    page_size: int

class ModelStat(BaseModel):
    model: str
    count: int
    avg_latency: float
    total_tokens: int
    success_rate: float

class AIStatsOut(BaseModel):
    total_requests: int
    success_rate: float
    total_tokens: int
    avg_latency: float
    models: List[ModelStat]

class CleanMetricsIn(BaseModel):
    days: int = 30

File: frontend/src/types/ai-monitor.ts
export interface AIMetric {
  id: string;
  timestamp: string;
  module?: string;
  model: string;
  provider: string;
  latency: number;
  prompt_tokens: number;
  completion_tokens: number;
  total_tokens: number;
  status: "success" | "error";
  error_message?: string;
}

export interface PaginatedAIMetrics {
  items: AIMetric[];
  total: number;
  page: number;
  page_size: number;
}

export interface ModelStat {
  model: string;
  count: number;
  avg_latency: number;
  total_tokens: number;
  success_rate: number;
}

export interface AIStats {
  total_requests: number;
  success_rate: number;
  total_tokens: number;
  avg_latency: number;
  models: ModelStat[];
}

export interface MetricFilters {
  page: number;
  page_size: number;
  module?: string;
  model?: string;
  provider?: string;
  status?: string;
  start_time?: string;
  end_time?: string;
}
